### PR TITLE
Bug fix/cleanup 1705

### PR DESCRIPTION
**Pull request by jsteemann**

### CODE REVIEW

File: arangod/Aql/Functions.cpp [hunk 1]
```diff
@@ -112,359 +114,399 @@ static_assert(DateSelectionModifier::WEEK < DateSelectionModifier::MONTH,
 static_assert(DateSelectionModifier::MONTH < DateSelectionModifier::YEAR,
               "incorrect date selection order");
 
+typedef void(*format_func_t)(std::string& wrk, tp_sys_clock_ms const&);
+typedef std::unordered_map<std::string,format_func_t> dateMapType;
+std::unordered_map<std::string,format_func_t> dateMap;
+auto unix_epoch = date::sys_seconds{seconds{0}};
 
-/*
```
Comment (dothebart) [inline arangod/Aql/Functions.cpp]: please leave this list in place at the start of the file, or move it else where (like a README) so people starting new aql functions can find it.

File: arangod/Aql/Functions.cpp [hunk 2]
```diff
@@ -112,359 +114,399 @@ static_assert(DateSelectionModifier::WEEK < DateSelectionModifier::MONTH,
...
-  - UnsetOrKeep ()
-  - GetDocumentByIdentifier ()
-  - MergeParameters()
-  - FlattenList()
+std::regex theDateFormatRegex;
 
-- now do your work with the parameters
-- build ub a result using a VPackbuilder like you would with regular velocpyack.
-- return it wrapping it into an AqlValue
+std::string regexReplace(std::string const& search,
```
Comment (dothebart) [inline arangod/Aql/Functions.cpp]: If we want to keep the generic name, we should move this out to i.e. basics, and replace the `dateMapType` with some generic name, and make tp a template parameter.
Else drop this generic name for a rather special purpose one.

**Comment by jsteemann** [comment]
@dothebart : changes implemented.

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/495/

**Pull request merged**